code/Controllers/HomeController.cs

```diff
@@ -45,17 +46,42 @@ public HomeController(BagsApi bagsApi)
 				if (product?.Images != null && product.Images.Count() > 0)
 					model.Images = product.Images
 						.OrderBy(image => image.Priority)
-						.SelectAndSwallowReferenceType(image =>new Uri(image.Large))
+						.SelectAndSwallowReferenceType(image => new Uri(image.Large))
 						.Where(image => image != null);
 
 				if (product?.Tags != null && product.Tags.Count() > 0)
 				{
 					var brand = product.Tags.FirstOrDefault(x => x.IsBrand)?.TagName ?? "";
-					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x=> x.TagName).ToArray());
+					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x => x.TagName).ToArray());
 
 					model.Title = $"{brand} {product.Name} - {styles}";
 				}
 			}
+			else if (!String.IsNullOrEmpty(tagId))
+			{
+				model.Url = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+
+				var tags = tagId.Split('_');
+				var products = await bagsApi.GetProductsByTags(tags);
+
+				if (products?.Count > 0)
+					model.Images = products.Take(5)
+						.Select(product => product.Images
+							.OrderBy(image => image.Priority)
+							.SelectAndSwallowReferenceType(image => new Uri(image.Large))
+							.Where(image => image != null).FirstOrDefault());
+
+
+				var tagModels = await bagsApi.GetTags();
```

Comment:
var productsTask = bagsApi.GetProductsByTags(tagList);
                var taskTagModels = bagsApi.GetTags();

                var products = await productsTask;
                if (products?.Count > 0)
                    model.Images = (await productsTask ?? Enumerable.Empty<BagsApi.Product>())
                        .Take(5)
                        .Select(product => product.Images
                                .OrderBy(image => image.Priority)
                                .SelectAndSwallowReferenceType(image => new Uri(image.Large))
                                .Where(image => image != null)
                                .FirstOrDefault());

Avoiding this if statement will bring empty enumeration if there are no products available. This will update Model.Images to be replaced as well. which should not be the case